configure.ac

```diff
@@ -51,6 +51,7 @@ LIBS="$LIBS -lcuda"
 AC_CHECK_FUNCS([[cuGetErrorName] [cuGetErrorString]])
 AC_CHECK_DECLS([[cuGetErrorName], [cuGetErrorString]],
   [], [], [[#include <cuda.h>]])
+AC_CHECK_HEADERS(unistd.h sys/stat.h)
```

Comment:
@jakubjelinek, turns out that's not the best position for the `AC_CHECK_HEADERS`: at this point, `-lcuda` is effective, and we'll get: `checking for ANSI C header files... no` if building nvptx-tools in a configuration where that's not available:

    configure:3794: checking for ANSI C header files
    configure:3814: gcc -c -g -O2   conftest.c >&5
    configure:3814: $? = 0
    configure:3887: gcc -o conftest -g -O2     conftest.c  -lcuda >&5
    /usr/bin/ld: cannot find -lcuda
    collect2: error: ld returned 1 exit status
    configure:3887: $? = 1
    configure: program exited with status 1
    [...]
    configure:3898: result: no

... or, in a configuration where building against the stub libcuda included in the CUDA toolkit (`--with-cuda-driver=[...]/cuda-6.5.14 --with-cuda-driver-lib=[...]/cuda-6.5.14/lib64/stubs`), the execution test fails:

    configure:3794: checking for ANSI C header files
    configure:3814: gcc -c -g -O2 -I[...]/cuda-6.5.14/include  conftest.c >&5
    configure:3814: $? = 0
    configure:3887: gcc -o conftest -g -O2 -I[...]/cuda-6.5.14/include  -L[...]/cuda-6.5.14/lib64/stubs  conftest.c  -lcuda >&5
    configure:3887: $? = 0
    configure:3887: ./conftest
    ./conftest: error while loading shared libraries: libcuda.so.1: cannot open shared object file: No such file or directory
    configure:3887: $? = 127
    configure: program exited with status 127
    [...]
    configure:3898: result: no

I'm testing the obvious fix.

Reply:
Resolved in commit b526ca4130c47424927da8f8005aa85ea51ea9b4 "Make sure that '-lcuda' etc. aren't in effect during generic checks [#18]".